spacewalk/certs-tools/rhn_bootstrap_strings.py

```diff
@@ -555,6 +553,93 @@ def getRegistrationStackSh(saltEnabled):
   zypper --non-interactive up {PKG_NAME_UPDATE} $RHNLIB_PKG ||:
 fi
 
+if [ "$INSTALLER" == apt ]; then
+    function check_deb_pkg_installed {{
+        dpkg-query -W -f='${{Status}}' $1 2>/dev/null | grep -q "ok installed"
+    }}
+
+    function getA_CLIENT_CODE_BASE() {{
+        local BASE=""
+        local VERSION=""
+
+	if [ -f /etc/os-release ]; then
+	    BASE=$(source /etc/os-release; echo $ID)
+	    VERSION=$(source /etc/os-release; echo $VERSION_ID)
+        fi
+        A_CLIENT_CODE_BASE="${{BASE:-unknown}}"
+	local VERCOMPS=(${{VERSION/\./ }}) # split into an array 18.04 -> (18 04)
+        A_CLIENT_CODE_MAJOR_VERSION=${{VERCOMPS[0]}}
+	A_CLIENT_CODE_MINOR_VERSION=$((${{VERCOMPS[1]}} + 0)) # convert "04" -> 4
+    }}
+
+    function getA_MISSING() {{
+        local NEEDED="salt-common salt-minion"
+        A_MISSING=""
+        for P in $NEEDED; do
+            check_deb_pkg_installed "$P" || A_MISSING="$A_MISSING $P"
+        done
+    }}
+
+    function test_deb_repo_exists() {{
+      local repourl="$CLIENT_REPO_URL"
+
+      $FETCH $repourl/dists/bootstrap/Release
+      if [ ! -f "Release" ] ; then
+        echo "Bootstrap repo '$repourl' does not exist."
+        repourl=""
+        CLIENT_REPO_URL=""
+      fi
+      rm -f Release
+    }}
+
+    function setup_deb_bootstrap_repo() {{
+      local repopath="$CLIENT_REPO_FILE"
+      local repourl="$CLIENT_REPO_URL"
+
+      test_deb_repo_exists
+
+      if [ -n "$CLIENT_REPO_URL" ]; then
+        echo " adding client software repository at $repourl"
+        echo "deb [trusted=yes] $repourl bootstrap main" >"$repopath"
+      fi
+    }}
+
+    echo "* check for necessary packages being installed..."
+    getA_CLIENT_CODE_BASE
+    echo "* client codebase is ${{A_CLIENT_CODE_BASE}}-${{A_CLIENT_CODE_MAJOR_VERSION}}.${{A_CLIENT_CODE_MINOR_VERSION}}"
+    getA_MISSING
+
+    CLIENT_REPOS_ROOT="${{CLIENT_REPOS_ROOT:-${{HTTPS_PUB_DIRECTORY}}/repositories}}"
+    CLIENT_REPO_URL="${{CLIENT_REPOS_ROOT}}/${{A_CLIENT_CODE_BASE}}/${{A_CLIENT_CODE_MAJOR_VERSION}}/${{A_CLIENT_CODE_MINOR_VERSION}}/bootstrap"
+    CLIENT_REPO_NAME="susemanager_bootstrap"
+    CLIENT_REPO_FILE="/etc/apt/sources.list.d/$CLIENT_REPO_NAME.list"
+
+    setup_deb_bootstrap_repo
+
+    apt-get --yes update
```

Comment:
Hmm, neiter for SLE nor for RHEL we update the system with the bootstrap script.
I think we should not do it with Ubuntu either.

Reply:
Btw, we still have a bug/fate open to allow the automatic update after bootstrap also for salt minions (since it was available for traditional clients in the past): https://bugzilla.suse.com/show_bug.cgi?id=1085493

Reply:
In the case of `apt-get` it's needed because repos don't get updated automatically when new repos are added/removed.

Reply:
does `apt-get --yes update` do a package update or is it something else ?

Reply:
It's the equivalent of `zypper refresh`.

Reply:
ahhhh, this is ok.